src/transpose.jl

```diff
@@ -31,19 +31,81 @@ Base.:(==)(A::AdjointMap, B::LinearMap)      = ishermitian(B) && A.lmap == B
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
+end
+@propagate_inbounds function mul!(y::AbstractMatrix, A::TransposeMap, x::AbstractMatrix, α::Number, β::Number)
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
+end
+# # AdjointMap
+@propagate_inbounds function mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector)
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x) : error("adjoint not implemented for $A")
+end
+@propagate_inbounds function mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix)
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x) : _generic_mapmat_mul!(y, A, x)
+end
+@propagate_inbounds function mul!(y::AbstractVecOrMat, A::AdjointMap, x::AbstractVector, α::Number, β::Number)
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x, α, β) : _generic_mapvec_mul!(y, A, x, α, β)
+end
+@propagate_inbounds function mul!(y::AbstractMatrix, A::AdjointMap, x::AbstractMatrix, α::Number, β::Number)
+    @boundscheck check_dim_mul(y, A, x)
+    @inbounds issymmetric(A.lmap) ? mul!(y, A.lmap, x, α, β) : _generic_mapmat_mul!(y, A, x, α, β)
+end
+# # ConjugateMap
+for (intype, outtype) in Any[Any[AbstractVector, AbstractVecOrMat], Any[AbstractMatrix, AbstractMatrix]]
+    @eval begin
+        @propagate_inbounds function mul!(y::$outtype, Ac::AdjointMap{<:Any,<:TransposeMap}, x::$intype)
+            @boundscheck check_dim_mul(y, Ac, x)
+            A = Ac.lmap
+            return @inbounds _conjmul!(y, A.lmap, x)
+        end
+        @propagate_inbounds function mul!(y::$outtype, Ac::AdjointMap{<:Any,<:TransposeMap}, x::$intype, α::Number, β::Number)
+            @boundscheck check_dim_mul(y, Ac, x)
+            A = Ac.lmap
+            return @inbounds _conjmul!(y, A.lmap, x, α, β)
+        end
+        @propagate_inbounds function mul!(y::$outtype, At::TransposeMap{<:Any,<:AdjointMap}, x::$intype)
+            @boundscheck check_dim_mul(y, At, x)
+            A = At.lmap
+            @inbounds isreal(A.lmap) ? mul!(y, A.lmap, x) : _conjmul!(y, A.lmap, x)
+        end
+        @propagate_inbounds function mul!(y::$outtype, At::TransposeMap{<:Any,<:AdjointMap}, x::$intype, α::Number, β::Number)
+            @boundscheck check_dim_mul(y, At, x)
+            A = At.lmap
+            return @inbounds _conjmul!(y, A.lmap, x, α, β)
+        end
+    end
+end
 
-Ac_mul_B!(y::AbstractVector, A::AdjointMap, x::AbstractVector) = A_mul_B!(y, A.lmap, x)
+# multiplication helper function
+@propagate_inbounds _conjmul!(y, A, x) = (mul!(y, A, conj(x)); conj!(y))
+@propagate_inbounds function _conjmul!(y, A, x::AbstractVector, α, β)
+    xca = rmul!(conj(x), conj(α))
```

Comment:
`conj!(x * α)` has the advantage that if `α` is e.g. complex and `x` is not, then it promotes.